src/pygeoprocessing/geoprocessing.py

```diff
@@ -1842,7 +1842,8 @@ def reclassify_raster(
 
     def _map_dataset_to_value_op(original_values):
         """Convert a block of original values to the lookup values."""
-        out_array = numpy.full(original_values.shape, target_nodata)
+        out_array = numpy.full(original_values.shape, target_nodata,
+                               dtype=numpy.float32)
```

Comment:
Since the datatype is user-defined, shouldn't this type be dependent on the input type? `_GDAL_TYPE_TO_NUMPY_LOOKUP` could be useful for this.

Reply:
I think it would come out the same because `raster_calculator` sets the final datatype, but it makes more sense to use the `target_datatype` parameter here too. Is that what you mean?

Reply:
Yes, I did mean that it would make sense to use the `target_datatype` parameter here as well, which I see is what you've added in 385243a.